src/Tree/TreeItem.tsx

```diff
@@ -75,51 +82,95 @@ const TreeItem: React.SFC<TreeItemProps> = ({
   label,
   color,
   onNodeClick,
+  onNodeDoubleClick,
   onNodeContextMenu,
   onRemove,
   hasChildren,
   isOpen,
   level,
   cursor,
   searchWords = [],
-}) => (
-  <Header
-    level={level}
-    onClick={onNodeClick}
-    onContextMenu={onNodeContextMenu}
-    highlight={Boolean(highlight)}
-    cursor={cursor}
-  >
-    {hasChildren &&
-      React.createElement(isOpen ? ChevronDownIcon : ChevronRightIcon, {
-        size: 11,
-        left: true,
-        color: "color.text.action",
-      })}
-    {tag && (
-      <NameTag condensed left color={color}>
-        {tag}
-      </NameTag>
-    )}
-    {icon &&
-      React.createElement(icon, {
-        size: 12,
-        color: iconColor || "color.text.lighter",
-        style: { marginLeft: 0, marginRight: 8, flex: "0 0 15px" },
-      })}
-    <Label hasChildren={hasChildren}>
-      <Highlighter
-        textToHighlight={label}
-        highlightStyle={{ color: constants.color.text.action, backgroundColor: "transparent", fontWeight: "bold" }}
-        searchWords={searchWords}
-      />
-    </Label>
-    {onRemove && (
-      <DeleteNode onClick={onRemove}>
-        <NoIcon size={12} />
-      </DeleteNode>
-    )}
-  </Header>
-)
+}) => {
+  const handleKeyDown = useCallback(
+    e => {
+      switch (e.key) {
+        case "Enter":
+          e.preventDefault()
+          if (e.altKey && onNodeContextMenu) {
+            onNodeContextMenu(e)
+            return
+          }
+
+          if (onNodeDoubleClick) {
+            onNodeDoubleClick(e)
+            return
+          }
+
+          if (onNodeClick) {
+            onNodeClick(e)
+          }
+          return
+        case " ":
+        case "Space": // the platform™
+          e.preventDefault()
+          if (onNodeClick) {
+            onNodeClick(e)
+          }
+          return
+        case "Delete":
+        case "Backspace":
+          e.preventDefault()
+          if (onRemove) {
+            onRemove(e)
+          }
+          return
+      }
+    },
+    [onNodeContextMenu, onNodeDoubleClick, onNodeClick],
```

Comment:
```suggestion
    [onNodeContextMenu, onNodeDoubleClick, onNodeClick, onRemove],
```

we need linter for this